lib/internal/modules/cjs/loader.js

```diff
@@ -223,13 +223,24 @@ ObjectDefineProperty(Module, 'wrapper', {
 function getModuleParent() {
   return moduleParentCache.get(this);
 }
+
+function setModuleParent(value) {
+  moduleParentCache.set(this, value);
+}
+
 ObjectDefineProperty(Module.prototype, 'parent', {
```

Comment:
moving this to be on the module itself is needed to make `delete` work

Reply:
Wouldn't that trigger a new deprecation warning for each `Module` instance?

```js
module.parent; // triggers one warning
module.parent?.parent; // triggers another warning?
```

Reply:
No, because the deprecation uses a deprecation code:
<https://github.com/nodejs/node/blob/ce84bacdc63017d8d7130b4f29dc78ac6ac138c8/lib/internal/util.js#L77-L82>